webapp/src/main/java/com/box/l10n/mojito/service/scheduledjob/jobs/ScheduledThirdPartySync.java

```diff
@@ -0,0 +1,146 @@
+import com.box.l10n.mojito.pagerduty.PagerDutyIntegrationService;
+import com.box.l10n.mojito.pagerduty.PagerDutyPayload;
+import com.box.l10n.mojito.quartz.QuartzPollableTaskScheduler;
+import com.box.l10n.mojito.service.pollableTask.PollableFuture;
+import com.box.l10n.mojito.service.scheduledjob.IScheduledJob;
+import com.box.l10n.mojito.service.scheduledjob.ScheduledJobRepository;
+import com.box.l10n.mojito.service.thirdparty.ThirdPartySyncJob;
+import com.box.l10n.mojito.service.thirdparty.ThirdPartySyncJobInput;
+import com.box.l10n.mojito.utils.ServerConfig;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.lang.text.StrSubstitutor;
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * Scheduled job that performs the third party sync for a repository off a cron schedule.
+ *
+ * @author mattwilshire
+ */
+@Component
+@DisallowConcurrentExecution
+public class ScheduledThirdPartySync implements IScheduledJob {
+
+  static Logger logger = LoggerFactory.getLogger(ScheduledThirdPartySync.class);
+
+  @Autowired private QuartzPollableTaskScheduler quartzPollableTaskScheduler;
+  @Autowired private ScheduledJobRepository scheduledJobRepository;
+  @Autowired private PagerDutyIntegrationService pagerDutyIntegrationService;
+
+  @Autowired ServerConfig serverConfig;
+
+  @Value(
+      "${l10n.scheduledJobs.thirdPartySync.notifications.title:MOJITO | Third party sync failed for {repository}}")
+  String notificationTitle;
+
+  private ScheduledJob scheduledJob;
+  private Long pollableTaskId;
+
+  @Override
+  public void execute(JobExecutionContext jobExecutionContext) throws JobExecutionException {
+    // Fetch the scheduled job and cast the properties
+    scheduledJob = scheduledJobRepository.findByJobKey(jobExecutionContext.getJobDetail().getKey());
+    ScheduledThirdPartySyncProperties scheduledJobProperties =
+        (ScheduledThirdPartySyncProperties) scheduledJob.getProperties();
+
+    logger.info(
+        "Third party sync for repository {} has started.", scheduledJob.getRepository().getName());
+
+    // Create ThirdPartySyncInput from scheduled job and properties
+    ThirdPartySyncJobInput thirdPartySyncJobInput =
+        new ThirdPartySyncJobInput(scheduledJob, scheduledJobProperties);
+
+    try {
+      PollableFuture<Void> task =
+          quartzPollableTaskScheduler.scheduleJobWithCustomTimeout(
+              ThirdPartySyncJob.class, thirdPartySyncJobInput, "thirdPartySync", 3600L);
+      pollableTaskId = task.getPollableTask().getId();
+      // Wait for sync to complete
+      task.get();
```

Comment:
I agree with it being strange to do a blocking get but this would have to be done either way, the polling will need to be done to ensure the job is finished to update the status.

RE: `thirdPartyService.syncMojitoWithThirdPartyTMS`. This may cause quite a few issues, the parent task creates child tasks to perform this sync in parallel (over multiple machines in a multi-clustered quartz environment) the call currently being used returns this PollableTask with all the child tasks which is also very useful for debugging why the task failed using the `api/pollableTasks` endpoint. I'm not entirely sure what the affects would be of not initially calling this sync inside the PollableTask context but I assume it would cause issues.

Reply:
@mattwilshire - good point around the child tasks tracking.
I don't have a better alternative atm, CC: @maallen if you have any thoughts of a better approach vs. needing to do the blocking get/wait. (Feel free to resolve this thread if not)

Reply:
`PollableTaskService` has a few methods to handle this waiting, ultimately you'll need to wait somewhere but I think we should be using our standardized 'wait for pollable task' logic rather here rather than just `get()`. You can also pass in a sleep time to using the `waitForPollableTask` methods rather than checking every 100ms for a job that runs over a period of minutes

Reply:
https://github.com/pinterest/mojito/blob/master/webapp/src/main/java/com/box/l10n/mojito/service/pollableTask/PollableTaskService.java#L160

Reply:
This is done @maallen, now using `waitForPollableTask `.